vscode/microsoft-kiota/src/extension.ts

```diff
@@ -157,27 +153,50 @@ export async function activate(
           );
           return;
         }
-        
+    
         const settings = getExtensionSettings(extensionId);
         workspaceGenerationType = config.generationType as string;
+        let result;
         switch (generationType) {
           case GenerationType.Client:
-            await generateClientAndRefreshUI(config, settings, outputPath, selectedPaths);
-          break;
+            result = await generateClientAndRefreshUI(config, settings, outputPath, selectedPaths);
+            break;
           case GenerationType.Plugin:
-            await generatePluginAndRefreshUI(config, settings, outputPath, selectedPaths);
-          break;
+            result = await generatePluginAndRefreshUI(config, settings, outputPath, selectedPaths);
+            break;
           case GenerationType.ApiManifest:
-            await generateManifestAndRefreshUI(config, settings, outputPath, selectedPaths);
-          break;
+            result = await generateManifestAndRefreshUI(config, settings, outputPath, selectedPaths);
+            break;
           default:
             await vscode.window.showErrorMessage(
               vscode.l10n.t("Invalid generation type")
             );
-          break;
+            return;
+        }
+        if (result && getLogEntriesForLevel(result, LogLevel.critical, LogLevel.error).length === 0) {
```

Comment:
+1 on this 

Do we also want an else clause that displays the reaons for the generation failure by picking the log entry that matches LogLevel.critical?

Reply:
Sure, we can display the error in the case of LogLevel.critical